test/integration/router_test.go

```diff
@@ -240,6 +273,146 @@ func TestRouter(t *testing.T) {
 	}
 }
 
+// TestRouterPathSpecificity tests that the router is matching routes from most specific to least when using
```

Comment:
@smarterclayton @rajatchopra - this is a test for the use case I was mentioning.  PTAL